### DISCUSSION TITLE
affichage de différent additionnelles

### DISCUSSION THREAD

**Discussion by SeddikRebbaj** [witsa/synapps#706] [Q&A]
Bonjour,

Je vous envoie ce message car je me suis retrouvé bloquer, je souhaite faire afficher sur le même acteur texte différentes valeur en fonction du bouton sur lequel on clique, vous retrouverez en pièce jointe un screen de mes additionnelles et de mon script .
![script](https://github.com/witsa/synapps/assets/151936804/4f45e481-e9c5-4ca6-aa60-cf154496f203)
![Additionnelles](https://github.com/witsa/synapps/assets/151936804/e8ba440c-ad0c-4385-ae4a-254faadd4851)

**Comment by Raphwit** [comment]
Bonjour, je vous propose des acteurs simples qui permet de changer le contenu d'un acteur texte en cliquant sur des boutons.

Ces boutons sont munis d'un script qui va se charger de récupérer l'acteur texte et de lui renseigner le contenu du bouton, évidemment, il est possible de lui renseigner n'importe quoi d'autre.

Je vous invite a vous inspirer de cette scène pour appréhender le fonctionnement du changement de contenu d'un acteur via script.

```
SYNAPPS-STUDIO-ACTOR|{"type":"layout/stack","key":"stack1","properties":{"fontSize":"5em"},"children":[{"type":"display/text","key":"text1","properties":{"content":"Text","textAlign":"center"}},{"type":"layout/stack","key":"stack2","properties":{"orientation":"horizontal"},"children":[{"type":"input/button","key":"button1","properties":{"content":"Test 1","textAlign":"center","horizontalAlignment":"expand"},"events":{"onClick":["context.getActor('text1').properties.content = this.properties.content;"]}},{"type":"input/button","key":"button2","properties":{"content":"Coucou","textAlign":"center","horizontalAlignment":"expand"},"events":{"onClick":["context.getActor('text1').properties.content = this.properties.content;"]}},{"type":"input/button","key":"button3","properties":{"content":"Synapps","textAlign":"center","horizontalAlignment":"expand"},"events":{"onClick":["context.getActor('text1').properties.content = this.properties.content;"]}}]}]}
```

**Comment by SeddikRebbaj** [comment]
J'ai testé, je vois que ça marche mais comment je dis au script que je veux faire afficher la valeur d'une ressources que j'ai sur mon wit ?

Cordialement

**Reply by Raphwit** [reply]
Pour cela, il suffit d'utiliser une additionnelle sur le bouton.

On effectuera une liaison de type "fournisseur de variable" sur cette additionnelle pour aller chercher la variable que l'on souhaite sur notre automate, il faudra bien penser à préciser dans la liaison quel champ on souhaite récupérer.

exemple de liaison : 

![image](https://github.com/witsa/synapps/assets/93767525/6ea9257f-15a7-4f7c-b8b8-e002241428ad)

il faudra également modifier un peu le script afin qu'il renseigne à l'acteur texte le contenu de l'additionnelle.

Le script devra donc ressemble à quelque chose de similaire : 
```
context.getActor('text1').properties.content = this.additionals.prop1;
```

"prop1" étant la clé de l'additionnelle créée précédemment.

Le bouton devrait maintenant renseigner à notre acteur texte la valeur de la ressource que l'on est aller chercher via l'additionnelle.

**Comment by SeddikRebbaj** [comment]
Dans mes l'additionnelles je n'ai pas fournisseur de variable j'ai seulement chemin vers variables et à cause de ça je ne peux pas sélectionner le fait que je veux afficher l'état, lui il comprends qu'il faut afficher set point mais pas la valeur du set point
![add](https://github.com/witsa/synapps/assets/151936804/ff6efc78-cdf3-448e-a29a-06d44dc3e59a)
![setpoint](https://github.com/witsa/synapps/assets/151936804/e204d3d1-6884-42aa-83ae-7989b59b771d)
 .

**Reply by Raphwit** [reply]
C'est le type de la liaison qui est "fournisseur de variable" et non pas le type de l'additionnelle.

L'additionnelle peut être de type "texte", cela convient bien à votre besoin.
